lib/cocoapods-core/source/aggregate.rb

```diff
@@ -60,6 +61,32 @@ def representative_set(name)
             if highest_version.nil? || (highest_version < source_version)
               highest_version = source_version
               representative_source = source
+
+            end
+          end
+        end
+        representative_source ? Specification::Set.new(name, representative_source) : nil
+      end
+
+      # Returns a set configured with the source which contains the all version in the aggregate.
+      #
+      # @param  [String] name
+      #         The name of the Pod.
+      #
+      # @return [Set] The most representative set for the Pod with the given
+      #         name. Returns nil if no representative source found containing a pod with given name.
+      #
+      def representative_set_for_search(name)
```

Comment:
this new method could use some test coverage

Reply:
Yeah i add unit test case for this method.